src/pages/ServiceList/ServiceListComponent.tsx

```diff
@@ -343,44 +343,41 @@ class ServiceListComponent extends React.Component<ServiceListComponentProps, Se
     }
 
     let serviceListComponent;
-    if (this.state.loading) {
-      serviceListComponent = <PfSpinner />;
-    } else {
-      serviceListComponent = (
-        <div>
-          <NamespaceFilter
-            initialFilters={[serviceNameFilter, istioFilter]}
-            onFilterChange={this.filterChange}
-            onError={this.handleError}
-          >
-            <Sort>
-              <Sort.TypeSelector
-                sortTypes={sortFields}
-                currentSortType={this.state.currentSortField}
-                onSortTypeSelected={this.updateSortField}
-              />
-              <Sort.DirectionSelector
-                isNumeric={this.state.currentSortField.isNumeric}
-                isAscending={this.state.isSortAscending}
-                onClick={this.updateSortDirection}
-              />
-            </Sort>
-            <RateIntervalToolbarItem
-              rateIntervalSelected={this.state.rateInterval}
-              onRateIntervalChanged={this.rateIntervalChangedHandler}
+    serviceListComponent = (
+      <div>
+        <PfSpinnerContainer />
```

Comment:
same comment as my other one - is it OK that we now always return the namespace filter here even if we are loading?  The old code did not return the namespace filter and only showed the spinner when currently loading.

Reply:
Yes, it's a smart component. Just place it wherever you want it and it looks at the Redux state for `isLoading` and determines if it needs to be rendered or not. And with the Axios interceptors hooked in, it knows whenever an HTTP request/response is made and updates the redux state automatically.
 This eliminates all of the client logic in multiple screens and consolidates it in the component itself.  For instance, all of this:
```
if (this.state.loading) {
  serviceListComponent = <PfSpinner />;
} else {
  serviceListComponent = (
    <div>
...
```
Is eliminated, and just becomes:
```
<PfSpinner />
```
It moves us from procedural logic in every `*Page` to a declarative smart component that you can place and it will work properly.